src/router_core/modules/mobile_sync/mobile.c

```diff
@@ -690,8 +690,24 @@ static void qcm_mobile_sync_on_mau_CT(qdrm_mobile_sync_t *msync, qd_parsed_field
             // Tell the python router about the new mobile sequence
             //
             qdr_post_set_mobile_seq_CT(msync->core, router->mask_bit, mobile_seq);
-        } else
-            qd_log(msync->log, QD_LOG_ERROR, "Received MAU from an unknown router");
+        } else {
+            //
+            // There is a possibility here that router_id is null but that is fine. We want to print it out either way
+            // which will help us in debugging.
+            //
+            char *r_id = 0;
+            bool free_rid = false;
```

Comment:
You don't need the free_rid thing.  It is safe to call 'free' on a null pointer.